new_app_setup.py

```diff
@@ -0,0 +1,266 @@
+import yaml
+import sys
+import os
+import subprocess
+import shutil
+import array
+from shutil import copyfile
+import argparse
+import tempfile
+from github import Github
+import smtplib
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from email.mime.base import MIMEBase
+from email import encoders
+from google.cloud import storage
+
+
+def main():
+
+    base_path = os.path.abspath(os.path.dirname(__file__))
+
+    parser = argparse.ArgumentParser(description='Usage: python3 '
+                                     'new_app_setup.py <Team name> '
+                                     '<List of users>')
+
+    parser.add_argument('team_name', type=str,
+                        help='Team name')
+
+    parser.add_argument('group_user', type=str, help='An email address that '
+                        'will be used for the entire group as a whole.')
+
+    parser.add_argument('-LOU', '--list_of_users',
+                        required=False,
+                        help='A list of the users for this team.')
+
+    parser.add_argument('-RAM', '--RAM', type=int, required=False,
+                        help='How many gigabytes of RAM your team is '
+                        'requesting, the deault is 2G.')
+
+    parser.add_argument('-CPU', '--CPU', type=int, required=False,
+                        help="How much CPU your team is requesting, the "
+                        "default is one.")
+
+    args = parser.parse_args()
+
+    team_name = args.team_name.replace(" ", "-").replace("_", "-").lower()
+
+    namespace_filename = team_name+"_namespace.yml"
+
+    pods_filename = team_name+"_pods.yml"
+
+    create_kubernetes_namespace_yml(namespace_filename, team_name)
+
+    create_kubernetes(namespace_filename, team_name, "namespace")
+
+    if args.RAM is not None:
+        ram_requested = args.RAM
+    else:
+        ram_requested = '2G'  # default
+
+    if args.CPU is not None:
+        cpu_requested = args.CPU
+    else:
+        cpu_requested = '1'  # default
+
+    create_kubernetes_pods_yml(pods_filename, team_name, ram_requested,
+                               cpu_requested)
+
+    create_kubernetes(pods_filename, team_name, "pods")
+
+    config_file = get_congif_yml_file(team_name)
+
+    path_to_congif_file = os.path.join(base_path, config_file)
+
+    send_config_email_to_users(team_name, args.group_user, config_file)
```

Comment:
Currently I think this does the following:

1. Create a namespace .yml file
2. Create a pod .yml file
3. Create the pod in k8's (but you do not create the namespace)
4. email the team their pod config

Eventually (not all in this PR), I think we want this to:

1. Create the namespace yml and actually create the namespace
2. Create a ResourceQuota yml and apply it to that namespace
3. Create k8's users/credentials for (1) the team and (2) the CI. These contain for `kubectl` to know how to find the cluster, etc. but, importantly, they also contain the cryptographic keys that are used to authenticate to k8's so we shouldn't email these
4. Set up a CI build including adding the keys as a secret
5. Set up 2 buckets
6. Create a gcloud account for the team so they can access those buckets
7. email the users some general instructions for using things, pointers to example .yml files for things they might want to do, etc.

I think we'll give team members a of copy of the team's k8's credentials and gcloud credentials via thumb drive or something since that's the actual secret they use to authenticate.

For this PR maybe just focus on (1) and (2)

Reply:
Ok, I will focus on those first two steps. In regards to making the namespace yml and actual namespace, is this not the case with these two lines?

    create_kubernetes_namespace_yml(namespace_filename, team_name)

    create_kubernetes(namespace_filename, team_name, "namespace")

My create_kubermetes function does the following:

    subprocess.check_call(['kubectl', 'create', '-f', filename])

and when I try "kubectl get namespaces" I see the ones that were created (which I need to go back and delete).

Reply:
yes, that looks right.